### PR TITLE
feat(Dataworker): Save merkle leaf proofs to Arweave

### PR DESCRIPTION
**Pull request by nicholaspai**
This is a really simple PR because `persistDataToArweave` was implemented so generically.

The disputer saves merkle leaf data to Arweave at the same opportunity that the disputer saves the bundle data to Arweave. This will double our Arweave usage but make proofs available for all to use.

Example data posted: https://ba6piv2daybktn3lp7fnzmzpaa2p5mz5tukzowxa6ohlji7k4fxq.arweave.net/CDz0V0MGAqm3a3_K3LMvADT-sz2dFZda4POOtKPq4W8

### CODE REVIEW

**Review by pxrl — COMMENTED** [review]
This looks good. We are however changing the format of the data submitted to Arweave; do we need to bump the `ARWEAVE_TAG_APP_VERSION` in SDK? cc @james-a-morris

**Comment by nicholaspai** [comment]
> This looks good. We are however changing the format of the data submitted to Arweave; do we need to bump the `ARWEAVE_TAG_APP_VERSION` in SDK? cc @james-a-morris 

Are we? This pr is saving a new file to arweave but not touching the existing file we upload. So now we're uploading 2 files per bundle rather than 1

File: src/dataworker/Dataworker.ts [hunk 1]
```diff
@@ -618,14 +618,46 @@ export class Dataworker {
...
+                ...leaf,
+                proof: expectedTrees.relayerRefundTree.tree.getHexProof(leaf),
+              };
+            }),
+            relayerRefundRoot: expectedTrees.relayerRefundTree.tree.getHexRoot(),
+            slowRelayLeaves: expectedTrees.slowRelayTree.leaves.map((leaf) => {
+              const { relayData, chainId, updatedOutputAmount } = leaf;
+              return {
+                ...leaf,
+                proof: expectedTrees.slowRelayTree.tree.getHexProof({ relayData, chainId, updatedOutputAmount }),
```
Comment (pxrl) [inline src/dataworker/Dataworker.ts:652]: Is there incidentally something additional in `leaf` that we don't want to persist to Arweave? Per the interface definition it should be possible to get away without this dereferencing.

```suggestion
              return {
                ...leaf,
                proof: expectedTrees.slowRelayTree.tree.getHexProof(leaf),
```

**Comment by pxrl** [comment]
> Are we? This pr is saving a new file to arweave but not touching the existing file we upload. So now we're uploading 2 files per bundle rather than 1

Sorry, you're totally right. There's no issue with this.

**Review by pxrl — APPROVED** [review]

**Review by james-a-morris — APPROVED** [review]
lgtm - this is a append-only change so we won't need a structure change

**Pull request merged**